source/fundamentals/builders/aggregates.txt

```diff
@@ -787,6 +787,47 @@ and ``temperature`` fields:
    :language: java
    :dedent:
 
+Densify
+-------
+
+Use the ``densify()`` method to create a
+`$densify <https://www.mongodb.com/docs/rapid/reference/operator/aggregation/densify/>`__
+pipeline stage that generates a sequence of documents to span a specified
+interval
+
+.. tip::
+
+   You can only use the ``$densify()`` aggregation stage when running
+   MongoDB v5.1 or later.
+
+The following example creates a pipeline stage that adds documents every
+15 minutes for which ``ts`` values do not already exist. The partition option
+specifies that the stage should group the documents using the ``position``
+field.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggregateDensify.java
+   :start-after: // begin densify aggregate
+   :end-before: // end densify aggregate
+   :language: java
+   :dedent:
+
+The following output highlights the documents added by the aggregate stage
+which contain``ts`` values every 15 minutes between the existing documents:
```

Comment:
Good suggestion. I think it could be even better with the order of sentences reversed in which you would want to introduce the snippet with "the following output..". The reasoning is that we should try not to separate "the following" with other sentences, admonitions, or content before what it describes.